### PR TITLE
Update accounts and balances to use entity.balance_timestamp

### LINKED ISSUE #6701: Use balance timestamp in REST API
### Problem

The new balance timestamp added in https://github.com/hashgraph/hedera-mirror-node/issues/6683 will provide a more accurate timestamp for when the balance was last updated.

### Solution

* Change `/api/v1/accounts` to use `entity.balance_timestamp`
* Change `/api/v1/accounts/{id}` to use `entity.balance_timestamp`
* Change `/api/v1/balances` to use `entity.balance_timestamp`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by edwin-greene**
The accounts and balances endpoints will now utilize the entity.balance_timestamp.

**Related issue(s)**:

Fixes #6701 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/__tests__/specs/balances/no-params.json
Comment (steven-sheehy) [inline hedera-mirror-rest/__tests__/specs/balances/no-params.json:1]: It's strange that this spec is called no-params yet has a test with the timestamp param.

File: hedera-mirror-rest/accounts.js [hunk 1]
```diff
@@ -408,31 +408,7 @@ const getOneAccount = async (req, res) => {
...
-    //Allow type coercion as the neValues will always be bigint and balanceFileTs may be a number
-    const timestampExcluded = neValues.some((value) => value == balanceFileTs);
-
-    if (timestampExcluded) {
-      throw new NotFoundError('Not found');
-    }
-
-    accountBalanceQuery.query = `ab.account_id = e.id and ab.consensus_timestamp = $${++paramCount}`;
-    accountBalanceQuery.params = [balanceFileTs ?? null];
+    accountBalanceQuery.query = `ab.account_id = e.id and ab.consensus_timestamp = e.balance_timestamp`;
```
Comment (steven-sheehy) [inline hedera-mirror-rest/accounts.js:411]: We can't join account_balance timestamp which is generated by 15 minute snapshot process with the balance_timestamp. We should only join on account_balance for when timestamp parameter is passed, in which case we should not change that flow.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/__tests__/specs/accounts/{id}/timestamp.json
[the remark this answers is not in the document]
Comment (edwin-greene) [reply hedera-mirror-rest/__tests__/specs/accounts/{id}/timestamp.json:1]: Thanks I have removed them.

File: hedera-mirror-rest/__tests__/specs/balances/no-params.json
Comment (edwin-greene) [reply hedera-mirror-rest/__tests__/specs/balances/no-params.json:1]: I removed the timestamp param test and the setup related to it except for a few items that demonstrated that the endpoint discerns between entries with older timestamps and entries with more recent ones.

File: hedera-mirror-rest/accounts.js (hunk 1 above)
Comment (edwin-greene) [reply hedera-mirror-rest/accounts.js]: Thanks I have removed this change.

File: hedera-mirror-rest/accounts.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ const processRow = (row) => {
       ? null
       : {
           balance: row.balance,
-          timestamp: utils.nsToSecNs(row.consensus_timestamp),
+          timestamp: utils.nsToSecNs(row.balance_timestamp),
```
Comment (steven-sheehy) [inline hedera-mirror-rest/accounts.js:45]: I don't see where we're removing the `select max(consensus_end) from record_file` sub-query. There might be multiple.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/accounts.js (hunk 2 above)
Comment (edwin-greene) [reply hedera-mirror-rest/accounts.js:45]: Thanks for spotting those, I have removed them.

**Review by xin-hedera — CHANGES_REQUESTED** [review]

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**